### ISSUE TITLE
a better error for  &[] (*[]*models.ServiceConfigVersionResponse) is not supported by the TextConsumer, can be resolved by supporting TextUnmarshaler interface

### ISSUE THREAD

**Issue by sershe** [go-swagger/go-swagger#1787]
## Problem statement

I'm trying to use a service with a swagger spec that looks to me like it was hand-created from an existing hand-written service (I find numerous discrepancies), and I'm trying to report issues in the spec for people to fix. Some are obvious; however, some response formatting issues (where the response almost, but not quite, matches the spec) seem to result in the following error, with different response types depending on the call: `&[] (*[]*models.ServiceConfigVersionResponse) is not supported by the TextConsumer, can be resolved by supporting TextUnmarshaler interface`.  The response from the server is 200 and includes the correct-ish response JSON. Either the JSON is not quite correct, or it's a generated code bug; for some classes I cannot see what the issue might be in json, esp. as it tends to be large.

Is it possible to improve the message to include what's actually wrong with the response? The spec is giant so having to implement unmarshalling for half the calls partially defeats the purpose of using the spec.

## Swagger specification

Attached.
[swagger.txt](https://github.com/go-swagger/go-swagger/files/2564107/swagger.txt)

## Steps to reproduce

Basically requires an Apache Ambari deployment. I generated the Go client using swagger and I'm calling the clusters/{clusterName}/configurations/service_config_versions API. I cannot paste the response fully but it follows this pattern:
`{
  "href" : "...",
  "items" : [
    {
      "href" : "...",
      "cluster_name" : "",
      "configurations" : [
        {
          "Config" : {
            "cluster_name" : "...",
            "stack_id" : "..."
          },
          "type" : "...",
          "tag" : "....",
          "version" : 1,
          "properties" : {
            "...." : "....",
          },
          "properties_attributes" : { }
        },` etc. 

## Environment
swagger version: 0.17.0
go version: 1.10.3  
OS: Mac on client, CentOS on server.
